src/averageAges.js

```diff
@@ -37,7 +47,17 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  let onlyWomen = people.filter(person => person.sex === 'f');
+
+  if (arguments.length === 2) {
+    onlyWomen = onlyWomen.filter(woman =>
+      people.some(child => child.mother === woman.name) === true);
```

Comment:
```suggestion
      people.some(child => child.mother === woman.name));
```
вот так работает точно так же